### ISSUE TITLE
how to build?

### ISSUE THREAD

**Issue by clarkewd** [carloop/can-utils-osx#1]
`gcc -o scx test.c` just seems to build the `test.c` file - can you provide details of how to build everything on mac?

**Comment by Gabrielcarvfer** [comment]
You probably need to run autogen.sh, then make, and finally you should be able to build your program with gcc -o scx test.c lib.o.

**Comment by rwohleb** [comment]
Multiple files, such as lib.c, are referencing files in the mac/* directory, which is not there. Looks like  there was an effort to clean up some of the files this past September and things were left in an unbuildable state. You just end up with "'mac/can.h' file not found" errors.

**Comment by alan707** [comment]
Hey! thanks for your interest! When I first started this project, I tried building on top of the OSX kernel and it is a bit more complicated than what I originally thought. In order for socketcan (and therefore can-utils) to work on OSX, we would need to create a Kernel Extension (kext) to port the socketcan network adapters into OSX. 

The simplest solution is to create a server that links the usb serial device to a virtual CAN network adapter in OSX.  Then we can modify can-utils to look at that virtual CAN network adapter. Will work on that today and see if I can make some headway.

**Comment by alan707** [comment]
I forgot to say, I left this repo as a placeholder and frankly forgot about it until you mentioned something! Time to maintain and update so it is usable.

**Comment by clarkewd** [comment]
Thanks Alan - it would be nice to have this working on OS X either via a kernel extension or a virtual network adapter. Let me know how I can help.

**Comment by pleasedodisturb** [comment]
If you could finish this and maintain - it would be wonderful.

Adding a homebrew installation flow would rocketjump the ease of installation.

**Comment by rageandqq** [comment]
> The simplest solution is to create a server that links the usb serial device to a virtual CAN network adapter in OSX. Then we can modify can-utils to look at that virtual CAN network adapter.

@alan707 Do you mind going into some more details on this? How would a server be linked to this USB serial device? If the server runs on OS X, how would it be able to interface with the USB device without the `socketcan` network adapters?

Really hoping that this project is able to be maintained. I'm struggling between being able to develop and deploy an application that requires `socketcan` and `can-utils` on both OS X and Linux environments. Thanks!

**Comment by fxb6476** [comment]
I just ran ./autogen.sh after pulling this and got these error messages. This is running on newest MAC osx.
$ ./autogen.sh 

----------
autoreconf
----------

./autogen.sh: line 19: glibtoolize: command not found
configure.ac:19: installing 'config/autoconf/compile'
configure.ac:9: installing 'config/autoconf/config.guess'
configure.ac:9: installing 'config/autoconf/config.sub'
configure.ac:22: installing 'config/autoconf/install-sh'
configure.ac:22: installing 'config/autoconf/missing'
GNUmakefile.am:25: error: Libtool library used but 'LIBTOOL' is undefined
GNUmakefile.am:25:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
GNUmakefile.am:25:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
GNUmakefile.am:25:   If 'LT_INIT' is in 'configure.ac', make sure
GNUmakefile.am:25:   its definition is in aclocal's search path.
GNUmakefile.am: installing 'config/autoconf/depcomp'
autoreconf: automake failed with exit status: 1

How can I be of help to fix this? I need this working for a school project.

**Comment by kiserem** [comment]
It looks like your best bet, fxb6476 will be to go get virtual box, install debian on it, and then run apt-get to get can_utils on your virtual box.  I poked a little at this, and it doesn't look like this will work for you... and if it does it'll be a lot of headache.

**Comment by fxb6476** [comment]
Well that sucks! I had herd before that using a virtual machine can cause connectivity issues with the CAN but I will experiment with this further thank you!

**Comment by meatherly** [comment]
@fxb6476 I had to install some stuff with brew `brew install automake autoconf libtool` then `autogen.sh` worked.

**Comment by RoninDusette** [comment]
I find it silly that you made this public and neglected it, wrote not proper documentation, and also obviously did not test it, because it does not build as it stands. lol. For shame.

**Comment by alan707** [comment]
> @fxb6476 I had to install some stuff with brew `brew install automake autoconf libtool` then `autogen.sh` worked.

@meatherly did that work on an OSX machine?

**Comment by alan707** [comment]
> I find it silly that you made this public and neglected it, wrote not proper documentation, and also obviously did not test it, because it does not build as it stands. lol. For shame.

@RoninDusette I opened sourced this project to see if people might be able
to help out to make it work since I was stuck. Do you have any recommendations?

**Comment by RoninDusette** [comment]
Yeah; add documentation. If you started developing it, the usual method if you are going to version it, is don't push any functionality to the master branch unless it works, which this obviously doesn't. I mean, if you had it in a dev branch, I would get it. But this:

"the popular Linux based can-utils and SocketCAN now available for OS X"

Is very misleading, as you obviously didn't bring it to mac, you have a stub for a test, and zero documentation. It's almost like you just copy/pasted the source, slapped it in a new repo under your name, and expected the community to do the heavy lifting. 

Please update your README, so that it's obvious that you are stuck and would like to get help to get it up and running. I will be more than willing to help out. I just need to know what you did, how you even possibly tried to get it working, caveats, etc. I am sure everyone would like to know.

@meatherly  Yo; mind elaborating I did that same thing and it wouldn't compile. Can you give us some insight? Mac version, kernel modules enabled, and symlinking or silly things you have to do to get it working? 

If I get it working I am just going to modify it and open source it under my name, as it's obviously just the source code with very little modification, if that, and it doesn't work. haha

**Comment by RoninDusette** [comment]
So, nothing from any of you? I got it compiled and almost working. And I don't even really know C. I will do the legwork, and just post it on my github under my authorship, considering everyone else wants to just let it sink.

**Comment by domints** [comment]
@RoninDusette and yet, you didn't. You created a fork and you changed nothing. I find that silly that you decided to be so rude, show off so much and then do nothing. Shame on you, Josh. 

It's Open Source - Alan created what he created, did what he could and when he stopped - stopped caring, stopped having time, stopped at a wall he hit. And you decided to be rude because you couldn't just come and use his work as is. 

Well, tough luck mate, you can take what he made, thank him and use it for your purpose, you can encourage him to fix it or help fixing. But being rude about how Alan uses his free time shouldn't be an option in civilised world.

@alan707 Did you make any progress? Unfortunately I am not that good in C++ to be able to efficiently resolve any issues that might be coming during the build.

**Comment by FernandoSVasconcelos** [comment]
Is there any trick to use the Linux can libs in a Mac with the new apple silicon chips ?
